src/pmdas/bpf/modules/netatop.c

```diff
@@ -0,0 +1,473 @@
+static struct netatop_bpf *obj;
+static char **instid_strings;
+static pmdaIndom *netatop_indoms;
+static unsigned int indom_id_mapping[INDOM_COUNT];
+
+
+#define METRIC_COUNT 8
+enum metric_name {
+  TCPSNDPACKS,
+  TCPSNDBYTES,
+  TCPRCVPACKS,
+  TCPRCVBYTES,
+  UDPSNDPACKS,
+  UDPSNDBYTES,
+  UDPRCVPACKS,
+  UDPRCVBYTES };
+enum metric_indom { NETATOP_INDOM };
+
+/*
+bcc.proc.net.tcp.send.calls
+bcc.proc.net.tcp.send.bytes
+bcc.proc.net.tcp.recv.calls
+bcc.proc.net.tcp.recv.bytes
+bcc.proc.net.udp.send.calls
+bcc.proc.net.udp.send.bytes
+bcc.proc.net.udp.recv.calls
+bcc.proc.net.udp.recv.bytes
+*/
+
+char* metric_names[METRIC_COUNT] = {
+    [TCPSNDPACKS] = "proc.net.tcp.send.packs",
+    [TCPSNDBYTES] = "proc.net.tcp.send.bytes",
+    [TCPRCVPACKS] = "proc.net.tcp.recv.packs",
+    [TCPRCVBYTES] = "proc.net.tcp.recv.bytes",
+    [UDPSNDPACKS] = "proc.net.udp.send.packs",
+    [UDPSNDBYTES] = "proc.net.udp.send.bytes",
+    [UDPRCVPACKS] = "proc.net.udp.recv.packs",
+    [UDPRCVBYTES] = "proc.net.udp.recv.bytes",
+};
+
+char* metric_text_oneline[METRIC_COUNT] = {
+    [TCPSNDPACKS] = "tcp packets sent",
+    [TCPSNDBYTES] = "tcp bytes sent",
+    [TCPRCVPACKS] = "tcp packets received",
+    [TCPRCVBYTES] = "tcp bytes received",
+    [UDPSNDPACKS] = "udp packets sent",
+    [UDPSNDBYTES] = "udp bytes sent",
+    [UDPRCVPACKS] = "udp packets received",
+    [UDPRCVBYTES] = "udp bytes received"
+};
+
+char* metric_text_long[METRIC_COUNT] = {
+    [TCPSNDPACKS] = "tcp packets sent (tracepoint/sock/sock_send_length)",
+    [TCPSNDBYTES] = "tcp bytes sent (tracepoint/sock/sock_send_length)",
+    [TCPRCVPACKS] = "tcp packets received (tracepoint/sock/sock_recv_length)",
+    [TCPRCVBYTES] = "tcp bytes received (tracepoint/sock/sock_recv_length)",
+    [UDPSNDPACKS] = "udp packets sent (tracepoint/sock/sock_send_length)",
+    [UDPSNDBYTES] = "udp bytes sent (tracepoint/sock/sock_send_length)",
+    [UDPRCVPACKS] = "udp packets received (tracepoint/sock/sock_recv_length)",
+    [UDPRCVBYTES] = "udp bytes received (tracepoint/sock/sock_recv_length)"
+};
+
+static unsigned int netatop_metric_count(void)
+{
+    return METRIC_COUNT;
+}
+
+static char* netatop_metric_name(unsigned int metric)
+{
+    return metric_names[metric];
+}
+
+static unsigned int netatop_indom_count(void)
+{
+    return INDOM_COUNT;
+}
+
+static void netatop_set_indom_serial(unsigned int local_indom_id, unsigned int global_id)
+{
+    pmNotifyErr(LOG_INFO, "%s", __FUNCTION__);
+    indom_id_mapping[local_indom_id] = global_id;
+}
+
+static int netatop_metric_text(int item, int type, char **buffer)
+{
+    if (type & PM_TEXT_ONELINE) {
+        *buffer = metric_text_oneline[item];
+    } else {
+        *buffer = metric_text_long[item];
+    }
+
+    return 0;
+}
+
+
+static void netatop_register(unsigned int cluster_id, pmdaMetric *metrics, pmdaIndom *indoms)
+{
+    pmNotifyErr(LOG_INFO, "%s Adding netatop metrics", __FUNCTION__);
+    /* bpf.netatop.tcpsndpacks */
```

Comment:
The metric name in the comment (here and below) doesn't match the namespace used - I guess this changed over time?